README.md

```diff
@@ -1,19 +1,152 @@
+2. [Adding Questions](/README.md#adding-questions)
+3. [Managing Questions](/README.md#managing-questions)
+4. [Updating local database](/README.md#updating-local-database)
+5. [Downloading questions](/README.md#downloading-questions)
+6. [References](/README.md#references)
+
+# Introduction
+
+Abbreviations: MXML (Moodle XML)
+
+This plugin is used for creating quizzes that are unique for every student and contains questions with specific difficulties and topics chosen by evaluators.
+The selection of the questions is automated based on the options that evaluators choose, so you can have a good distribution of difficulty levels and topics.
+
+We want to choose the questions randomly based on the selected difficulty and topic, and to make sure that all the questions are used evenly.
+In order to do this consistently when reusing the same questions in multiple Moodle instances, we need to keep timestamps of every question last use.
+
+As defined in [Managing Questions](README.md#managing-questions), the `last_used` question tag keeps the time of the question was last use, measured in seconds since the Epoch (00:00:00 UTC, January 1, 1970).
+The plugin automatically updates the `last_used` tag after every questions use, and can [export it in multiple ways](/README.md#downloading-questions), so it can be imported and reused in a different Moodle instance.
+
+# Installation
+
+### Installation From Downloaded zip File
+
+You can download the zip from:
+#### latest [(master branch)](https://github.com/systems-cs-pub-ro/quiz-manager-moodle/zipball/master)
+
+Follow the steps below in your Moodle installation (administrative rights are required):
+
+* Go to `Site administration > Plugins > Install plugins`.
+* Upload the zip file. If you are prompted to add extra details, under `Plugin type` select `Question type (qtype)`.
+* Go to `Show more > Rename the root directory` and type `quizmanager`.
+* If your target directory is not writable, you will see a warning message. Proceed to [Manual installation](/README.md#manual-installation-from-zip-file).
+* Press `Install plugin from the ZIP file` button.
+
+### Manual Installation From zip file
+
+If you are prompted with the message `Plugin type location /your/moodle/path/question/type is not writable`, you have to copy the files manually to the server file system, using a zip file or [Git](/README.md#manual-installation-using-git).
+
+* Download the [zip file](https://github.com/systems-cs-pub-ro/quiz-manager-moodle/zipball/master).
+* Unzip the file in the correct location: `/your/moodle/path/question/type`.
+* Rename the directory to `quizmanager`.
+
+### Manual Installation Using Git
+
+To install using Git, use this command:
+
+```
+git clone https://github.com/systems-cs-pub-ro/quiz-manager-moodle.git /your/moodle/path/question/type/quizmanager
+```
+
+# Adding Questions
+
+Questions are added using the Moodle interface, after the plugin was installed, from the quiz menu.
```

Comment:
Where are questions added? To the question database? How is that different from importing questions? Should this be importing questions?

Or is this adding questions to a quiz?

Reply:
It's about adding to a quiz.
```suggestion
Questions are added to a quiz using the Moodle interface, after the plugin was installed, from the quiz menu.
**Note:** You have to import questions to question bank beforehand.
```